tools/ci/templates/steps-docs.yaml

```diff
@@ -0,0 +1,36 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+
+# [TEMPLATE] Build the project documentation
+
+steps:
+  - task: Powershell@2
+    displayName: 'Generate docs'
+    inputs:
+      targetType: filePath
+      filePath: docs/generateDocs.ps1
+      workingDirectory: docs
+      failOnStderr: false
+
+  - task: PublishPipelineArtifact@0
+    displayName: 'Publish as artifact'
+    inputs:
+      artifactName: 'docs'
+      targetPath: 'build/docs/generated'
+
+    # Push docs if enabled, build step succeeded, and was not triggered by a PR
+  - task: PowerShell@2
+    displayName: 'Push docs to git if needed'
+    condition: |
+      and(
+        variables['publish'],
+        succeeded(),
+        not(eq(variables['Build.Reason'], 'PullRequest'))
```

Comment:
There seem to lots of other Build.Reasons. e.g. "CheckInShelveset: Gated check-in trigger." So it seems quite broad to check "not PullRequest"?

Reply:
Will remove, this pipeline is not even triggered by PRs.